astroquery/cadc/core.py

```diff
@@ -237,6 +239,136 @@ def get_collections(self):
+            Just return the dict of HTTP request parameters.
+        show_progress: bool, optional
+            Whether to display a progress bar if the file is downloaded
+            from a remote server.  Default is `True`.
+
+        Returns
+        -------
+        list : A list of `~astropy.io.fits.HDUList` objects
+        """
+
+        request_payload = self._args_to_payload(coordinates=coordinates,
+                                                radius=radius,
+                                                collection=collection)
+        request_payload['query'] = request_payload['query'] + " AND (dataProductType = 'image')"
+
+        if get_query_payload:
+            return request_payload
+
+        query = request_payload['query'] + " AND (dataProductType = 'image')"
+        response = self.run_query(query, operation='sync')
+        query_result = response.get_results()
+
+        if query_result and len(query_result) == 2000:
+            logger.debug("Synchronous query results capped at 2000 results - results may be truncated")
+
+        images_urls = self.get_image_list(query_result, coordinates, radius)
+        images = []
+
+        try:
+            readable_objects = [commons.FileContainer(url, encoding='binary',
+                                                      show_progress=show_progress) for url in images_urls]
+            for obj in readable_objects:
+                images.append(obj.get_fits())
+        except HTTPError as err:
+            logger.debug(
+                "{} - Problem retrieving the file: {}".format(str(err), str(err.url)))
+            pass
+
+        return images
+
+    @class_or_instance
+    def get_image_list(self, query_result, coordinates, radius):
+        """
+        Function to map the results of a CADC query into URLs to
+        corresponding data and cutouts that can be later downloaded.
+
+        The function uses the IVOA DataLink Service
+        (http://www.ivoa.net/documents/DataLink/) implemented at the CADC.
+        It works directly with the results produced by Cadc.query_region and
+        Cadc.query_name but in principle it can work with other query
+        results produced with the Cadc query as long as the results
+        contain the 'caomPublisherID' column. This column is part of the
+        caom2.Plane table.
+
+        Parameters
+        ----------
+        query_result : result returned by Cadc.query_region() or
+                    Cadc.query_name(). In general, the result of any
+                    CADC TAP query that contains the 'caomPublisherID' column
+                    can be use here.
+        coordinates : str or `astropy.coordinates`.
+            Coordinates around which to query
+        radius : str or `astropy.units.Quantity`.
+            The radius of the cone search AND cutout area
+
+        Returns
+        -------
+        A list of URLs to data.
+        """
+
+        def chunks(obj_list, chunk_len):
```

Comment:
For brevity, I would replace the function with the oneliner:
`(publisher_ids[pos:pos + batch_size] for pos in range(0, len(publisher_ids), batch_size))` with a short comment at the beginning of the block.

Reply:
mmm, much cleaner!